text/0000-move-analyze-phase.md

```diff
@@ -0,0 +1,116 @@
+* gid
+* uid
+* log-level
+* Path to output analyzed.toml
+* Layers Directory
+* Skip Layers
+* Daemon
+
+## Output
+
+* Exit status
+* Info-level logs to `stdout`
+* Error-level logs to `stderr`
+* Analysis metadata [`analyzed.toml`](https://github.com/buildpacks/spec/blob/main/platform.md#analyzedtoml-toml), including run-image information.
+* Buildpacks (derived from inline buildpacks in project descriptor, or buildpacks in project descriptor that are not present in the builder)
+* Buildpacks order [`order.toml`](https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml)
+* Lifecycle configuration (derived from configuration in project descriptor)
+* Mutated app source code (applying include and exclude rules in project descriptor)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+A platform MUST execute the analyze phase either by invoking the `/cnb/lifecycle/analyzer` binary or by executing `/cnb/lifecycle/creator`.
+
+The `analyzer` binary will have access to the [`Keychain`](https://github.com/buildpacks/lifecycle/blob/main/auth/env_keychain.go), and MUST NOT execute arbitrary code provided by either the buildpack user or buildpack author.
+
+The [logic in the `analyzer` phase that reads image metadata and outputs an `analyzed.toml`](https://github.com/buildpacks/lifecycle/blob/main/analyzer.go#L34-L40) would be remain.
+
+The [logic in `pack` that parses a `project.toml`](https://github.com/buildpacks/pack/blob/main/project/project.go) would be copied or moved into the `analyzer`.
+
+The [logic in the `analyzer` phase that analyzes layers](hhttps://github.com/buildpacks/lifecycle/blob/main/analyzer.go#L54-L116) would be moved to the `restorer`. `restorer` already takes in `group.toml` as a flag.
+
+The app source code (which may be provided to the prepare either as a directory, volume, or tarball) would be mutated (either by copying it to a new location, or making changes directly). The `analyzer` may delete files to apply the include and exclude rules from `project.toml`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* Platform maintainers will need to update the order of their container execution and also update flags for `analyzer`, `detector`, and `restorer`.
+* Lifecycle will now take on the responsibility of processing `project.toml`
+
+# Alternatives
+[alternatives]: #alternatives
+
+- [Introduce Prepare Phase](https://github.com/buildpacks/rfcs/blob/4547fe1ce602877db24f09e5b08bc9713c979be0/text/0000-prepare-phase.md) (this same rfc, previous version)
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [Tekton prepare step](https://github.com/tektoncd/catalog/blob/11a17cfe87779099b0b61be3f1e496dfa79646b3/task/buildpacks-phases/0.1/buildpacks-phases.yaml#L61-L78)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Does `pack` still need to parse `project.toml`, or is there value in reading it early on (before lifecycle runs)?
+- Should we create a shared library for `project.toml` parsing?
```

Comment:
This is tied in to the general discussion about better exposing project libraries in pack: https://github.com/buildpacks/community/discussions/51. 

This formulation is interesting, though. Were you thinking of creating, in addition to imgutil, a bputil library that has common utilities for the CNB project?

Reply:
I think that `bputil` could live in the `pack` repo. I don't have a strong opinion on where the package lives. I sort of lean towards keeping it in `pack` but mostly because managing deps is annoying.

Reply:
Bumped to future work (all of project.toml)